pkg/controllers/hypershiftdeployment_controller.go

```diff
@@ -98,6 +98,16 @@ func (r *HypershiftDeploymentReconciler) Reconcile(ctx context.Context, req ctrl
 	configureInfra := hyd.Spec.Infrastructure.Configure
 	if configureInfra ||
 		(hyd.Spec.HostedClusterSpec != nil && hyd.Spec.HostedClusterSpec.Platform.Azure != nil) {
+		if hyd.Spec.Infrastructure.CloudProvider.Name == "" {
```

Comment:
To prevent error, I remember having to do `len(hyd.Spec.Infrastructure.CloudProvider) == 0`, or if fails in getting Name from a nil CloudProvider obj?

Reply:
@philipwu08 See the test log output above. This code I think is fine

Reply:
@philipwu08 I think the above code looks fine and using len() is essentially the same.